ext/hal/cmsis/Kconfig

```diff
@@ -1,8 +1,10 @@
-#
 # Copyright (c) 2016 Intel Corporation
-#
 # SPDX-License-Identifier: Apache-2.0
-#
 
-config HAS_CMSIS
+config HAS_CMSIS_CORE
 	bool
+	select HAS_CMSIS_CORE_M if CPU_CORTEX_M
```

Comment:
Hmm, wonder if this is better as 'default if CPU_CORTEX_M' instead of a select.

@ulfalizer thoughts?

Reply:
To elaborate, the idea here is that if `HAS_CMSIS_CORE` should automatically select an adequate CMSIS-Core type based on the CPU type.

For example, when CMSIS-Core(R) is added in the next PR, it would look something like this:
```
config HAS_CMSIS_CORE
	bool
	select HAS_CMSIS_CORE_M if CPU_CORTEX_M
	select HAS_CMSIS_CORE_R if CPU_CORTEX_R
```

The rationale for using `select` instead of `default` is as follows:

1. Since CMSIS-Core is a "driver" for CPU core, it makes very little sense to select a type that does not match the CPU type. For instance, if you have `CPU_CORTEX_M` selected, you would always want to select `HAS_CMSIS_CORE_M` which enables CMSIS-Core(M). For this reason, other config files should never need to override this.

2. Once `HAS_CMSIS_CORE` is selected, the right type of CMSIS-Core must be enabled. For example, it makes no sense to select `HAS_CMSIS_CORE` and then do `HAS_CMSIS_CORE_M=n` in another config file.

Reply:
This design solves another problem. Assume a vendor with all Cortex-A, Cortex-M and Cortex-R support. The vendor module will select HAS_CMSIS in the root module Kconfig file, without having to care about the different ARM variants. The right type will be selected based on the actual CPU variant. 

So I like the solution by proposed here by @stephanosio .

Of course, these options must be used somewhere.